com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -346,6 +346,10 @@ private struct PacketLossCache
             public float PacketLoss;
         };
 
+        internal static event Action<UnityTransport> TransportInitialized;
+        internal static event Action<UnityTransport> TransportDisposed;
```

Comment:
Should a reference to the Transport be included in the disposed message? Is it still valid at that point?

Reply:
Yes, since we are using the Instance ID to track the adapter.

Reply:
Can you expand on what the value is of tracking dispose instead of shutdown? Does this need to be an event or can you use a weak reference?

To expand on Ian's concern and my other comment about internals vs public, at the point you have this being invoked the object is essentially "off limits" to other objects as far as best practices go - regardless of how this may be used right now this feels dangerous and dependent on tribal knowledge about the control flow. If all you need is the ID, why not pass that as the event parameter instead of the entire (mostly disposed) object?

Reply:
There's more to it. We are using the whole UnityTransport to access the Network Driver. Arguably, we could provide the Instance ID and the Network Driver only as arguments. I am not against this change. As for the Disposed, we definitely only need the Instance ID and I can change it in the same manner if you'd prefer that.